packages/react-testing/src/root.tsx

```diff
@@ -76,26 +77,34 @@ export class Root<Props> {
 
   act<T>(action: () => T, {update = true} = {}): T {
     let result!: T;
-
     if (this.acting) {
       return action();
     }
 
     this.acting = true;
 
-    withIgnoredReactLogs(() =>
-      act(() => {
-        result = action();
-      }),
-    );
+    const afterResolve = () => {
+      if (update) {
+        this.update();
+      }
+      this.acting = false;
 
-    if (update) {
-      this.update();
-    }
+      return result;
+    };
 
-    this.acting = false;
+    const promise = act(() => {
+      result = action();
+
+      if (isPromise(result)) {
+        return (result as unknown) as Promise<void>;
+      }
+    });
 
-    return result;
+    if (isPromise(result)) {
+      return Promise.resolve(promise).then(afterResolve) as any;
```

Comment:
Same request as above re: clarifying comments for this and the type casting of `act` above